packages/react-avatar/SPEC.md

```diff
@@ -0,0 +1,272 @@
+  'L7.91541 1.46793C8.44076 0.557947 9.39444 0 10.4245 0H25.5755C26.6056 0 27.5592 0.557951 28.0846 1.46793' +
+  'L35.5921 14.472C36.136 15.4141 36.136 16.5859 35.5921 17.528L28.0846 30.5321' +
+  'C27.5592 31.4421 26.6056 32 25.5755 32H10.4245C9.39443 32 8.44076 31.4421 7.91541 30.5321L0.407926 17.528Z"/>' +
+  '</svg>';
+```
+
+## API
+
+From [Avatar.types.tsx](https://github.com/microsoft/fluentui/blob/master/packages/react-avatar/src/components/Avatar/Avatar.types.tsx):
+
+### Slots
+
+- `image` - The Avatar's image, if available.
+- `label` - The text shown when there's no image. Defaults to the initials derived from `name` using `getInitials`.
+- `icon` - Icon displayed when there's no image or intials available.
+- `badge` - Badge to show the avatar's status.
+
+### Props
+
+```ts
+export interface AvatarProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /** The Avatar's image. */
+  image?: ShorthandValue<{}>;
+
+  /** The label shown when there's no image. Defaults to the initials derived from `name` using `getInitials`. */
+  label?: ShorthandValue<{}>;
+
+  /** Icon displayed when there's no image or intials available, or if `display="icon"`. */
+  icon?: ShorthandValue<{}>;
+
+  /** What the avatar displays. This can be used to override the default behavior, for example to show the icon even
+   *  if the avatar has initials that could be shown as the label. */
+  display?: 'image' | 'label' | 'icon';
+
+  /**
+   * Optional activity indicator
+   * * true: the avatar will be decorated according to activeDisplay
+   * * false: the avatar will be reduced in size and partially transparent
+   * * undefined (default): normal display
+   */
+  active?: boolean;
+
+  /**
+   * The type of visual treatment to use when `active="true"`
+   * @defaultvalue ring
+   */
+  activeDisplay?: 'ring' | 'shadow' | 'glow' | 'ring-shadow' | 'ring-glow';
+
+  /** Badge to show the avatar's status. */
+  badge?: ShorthandValue<BadgeProps>;
+
+  /** The name used for displaying the initials of the avatar if the image is not provided. */
+  name?: string;
+
+  /** The avatar can have a square shape. */
+  square?: boolean;
+
+  /** Size of the avatar */
+  size?: AvatarSizeValue;
+
+  /**
+   * If a non-standard size is needed, use customSize instead of size.
+   *
+   * The dimensions of the avatar will be the given customSize, and the icon
+   * and font sizes will be based on the next-smaller AvatarSizeValue.
+   * For more fine-grained control over the font and icon sizes, use tokens:
+   * `tokens={{ fontSize: '...px', fontWeight: '...', iconSize: '...px' }}`
+   */
+  customSize?: number;
+
+  /** Custom method for generating the initials from the name property, which is shown if no image is provided. */
+  getInitials?: (name: string, isRtl: boolean) => string;
+
+  /** Classes for the parts of the component */
+  classes?: { [key: string]: string };
+
```

Comment:
```suggestion
```

Please correct if I am wrong, but this should be done via slots:

```jsx
<Avatar label={{ className: 'foo' }} />
```

Reply:
Thanks for catching this. It was a holdover from dzearing's original prototype and can be removed.